fastlane_bot/helpers/txhelpers.py

```diff
@@ -117,578 +53,136 @@ def validate_and_submit_transaction(
+            except Exception as e:
+                self.cfg.logger.info(f"create_access_list({tx}) failed with {e}")
 
-        routes: the routes to be used in the transaction
-        src_amt: the amount of the source token to be sent to the transaction
-        gas_price: the gas price to be used in the transaction
+        tx["gas"] += self.cfg.DEFAULT_GAS_SAFETY_OFFSET
 
-        returns: the transaction function ready to be submitted
-        """
-        if self.ConfigObj.SELF_FUND:
-            return self.arb_contract.functions.fundAndArb(
-                routes, src_address, src_amt
-            )
-        return self.arb_contract.functions.flashloanAndArbV2(
-            flashloan_struct, routes
-        )
+        raw_tx = self.cfg.w3.eth.account.sign_transaction(tx, self.cfg.ETH_PRIVATE_KEY_BE_CAREFUL).rawTransaction
 
-    def _handle_function_build_exception(self, exception: Exception) -> int or None:
-        """Handles exceptions that occur during transaction building.
+        gas_cost_wei = tx["gas"] * gas_price
+        if self.cfg.network.GAS_ORACLE_ADDRESS:
+            gas_cost_wei += self.cfg.GAS_ORACLE_CONTRACT.caller.getL1Fee(raw_tx)
 
-        This method logs the exception and attempts to extract information from it. If the exception
-        indicates that the maximum fee per gas is less than the block base fee, it extracts and returns
-        the base fee. Otherwise, it logs a warning and returns None.
+        gas_cost_eth = Decimal(gas_cost_wei) / ETH_RESOLUTION
+        gas_cost_usd = gas_cost_eth * expected_profit_usd / expected_profit_gastkn
 
-        Args:
-            self: The instance of the class.
-            exception (Exception): The exception raised during transaction building.
+        gas_gain_eth = self.arb_rewards_portion * expected_profit_gastkn
+        gas_gain_usd = self.arb_rewards_portion * expected_profit_usd
 
-        Returns:
-            int or None: The base fee if it can be extracted from the exception, otherwise None.
+        self.cfg.logger.info(log_format(log_name="arb_with_gas", log_data={**log_object, "tx": tx}))
 
-        """
-        self.ConfigObj.logger.debug(
-            f"[helpers.txhelpers.build_transaction_with_gas] Error when building transaction: {exception.__class__.__name__} {exception}"
+        self.cfg.logger.info(
+            f"[helpers.txhelpers.validate_and_submit_transaction]:\n"
+            f"- Expected cost: {num_format(gas_cost_eth)} GAS token ({num_format(gas_cost_usd)} USD)\n"
+            f"- Expected gain: {num_format(gas_gain_eth)} GAS token ({num_format(gas_gain_usd)} USD)\n"
         )
-        if "max fee per gas less than block base fee" in str(exception):
-            message = str(exception)
-            return int_prefix(message.split("baseFee: ")[1])
 
+        if gas_gain_eth > gas_cost_eth:
+            self.cfg.logger.info("Attempting to execute profitable arb transaction")
+            try:
```

Comment:
`_send_private_transaction` can only be used on Ethereum.

All other chains + Tenderly require the normal transaction submission. 

Currently here only Tenderly uses the normal transaction submission & all other chains use the private method - which will fail except on Ethereum.

Reply:
Please suggest a robust non-hackie way for distinguishing between these two cases.

Reply:
OK, resolving;